### ISSUE TITLE
HTML invalid markup for accordion

### ISSUE THREAD

**Issue by MashaSinkevich** [chakra-ui/chakra-ui#7076]
### Description

To meet accessibility criteria "A" all HTML markup should be valid (see https://www.w3.org/WAI/WCAG21/quickref/#parsing).

During validation of our pages with https://validator.w3.org/nu/ we've found a bug from Chakra code (checked with https://chakra-ui.com/docs/components/accordion/usage, there is also `<div>` inside of `<button>`):

Error: Element [div](https://html.spec.whatwg.org/multipage/#the-div-element) not allowed as child of element [button](https://html.spec.whatwg.org/multipage/#the-button-element) in this context. (Suppressing further errors from this subtree.)

Possibly a change of `<div>` to `<span>` should help in this case and make this markup valid.

### Link to Reproduction

https://validator.w3.org/nu/

### Steps to reproduce

1. Go to ['...'](https://validator.w3.org/nu/)
2. Click on 'Check by input'
3. Add in the body section inside of textarea example from official Chakra UI website for accordion, e.g. full markup:
```
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>
<div class="chakra-accordion css-0"><div class="chakra-accordion__item css-17mg6aq"><h2><button type="button" id="accordion-button-:rb:" aria-expanded="true" aria-controls="accordion-panel-:rb:" class="chakra-accordion__button css-uttm9k" data-index="0"><div class="css-1eziwv">Section 1 title</div><svg viewBox="0 0 24 24" focusable="false" class="chakra-icon chakra-accordion__icon css-j2ph2z" aria-hidden="true"><path fill="currentColor" d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"></path></svg></button></h2><div class="chakra-collapse" style="overflow: hidden; display: block; opacity: 1; height: auto;"><div role="region" id="accordion-panel-:rb:" aria-labelledby="accordion-button-:rb:" class="chakra-accordion__panel css-1hb5ode">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</div></div></div><div class="chakra-accordion__item css-17mg6aq"><h2><button type="button" id="accordion-button-:re:" aria-expanded="false" aria-controls="accordion-panel-:re:" class="chakra-accordion__button css-uttm9k" data-index="1"><div class="css-1eziwv">Section 2 title</div><svg viewBox="0 0 24 24" focusable="false" class="chakra-icon chakra-accordion__icon css-186l2rg" aria-hidden="true"><path fill="currentColor" d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"></path></svg></button></h2><div class="chakra-collapse" style="overflow: hidden; display: none; opacity: 0; height: 0px;"><div role="region" id="accordion-panel-:re:" aria-labelledby="accordion-button-:re:" class="chakra-accordion__panel css-1hb5ode">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</div></div></div></div>
</body>
</html>
```
5. See 2 errors as from above.

### Chakra UI Version

2.4.2

### Browser

Google Chrome 108.0.5359.94 (Official Build) (x86_64)

### Operating System

- [X] macOS
- [X] Windows
- [X] Linux
